tests/cms/views/interactions/test_vote_action.py

```diff
@@ -0,0 +1,34 @@
+import pytest
+from django.contrib.messages.storage.fallback import FallbackStorage
+from django.test import RequestFactory
+
+from integreat_compass.cms.models import User, Vote
+from integreat_compass.cms.views.interactions import VoteFormView
+
+
+def get_request(offer_version_id):
+    request = RequestFactory().post(
+        "", {"approval": "False", "offer_version_id": offer_version_id}
+    )
+    request.user = User.objects.get(pk=1)
+    setattr(request, "session", "session")
+    messages = FallbackStorage(request)
+    setattr(request, "_messages", messages)
+    return request
+
+
+@pytest.mark.django_db
+def test_add_vote_if_not_exists(load_test_data):
+    assert not Vote.objects.filter(creator=1, offer_version=2).exists()
+    request = get_request(2)
+    VoteFormView.as_view()(request)
+    assert Vote.objects.get(creator=1, offer_version=2)
+    assert not Vote.objects.filter(creator=1, offer_version=2).first().approval
```

Comment:
```suggestion
    assert not Vote.objects.get(creator=1, offer_version=2).approval
```

Both of these lines do not really make sense, because of similar reasons in my previous comments.
- `assert Vote.objects.get()` does never run the `assert` statement, since there are only two possible outcomes of `get()`: An object (which always evaluates to `True`), or an exception (which lets the test fail before the `assert` even runs).
- Accessing attributes after `filter().first()` does not make sense since the only reason why you would prefer `filter().first()` over `get()` is when you want to tolerate `None` as results In this case, if `None` would be returned, this would throw an exception `AttributeError: 'NoneType' object has no attribute 'approval'`, which is a far worse error message than the `ObjectDoesNotExist` which is thrown by `get()` if no such object exists.

I'm not quite sure what part of my suggestion you don't like, could you explain a bit why you would prefer your solution? Am I missing something here?

Reply:
I thought I just did what you suggested, I just want to "assert" that the value of approval has been updated.